digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/lane/LaneDao.scala

```diff
@@ -0,0 +1,446 @@
+
+      case _ => None
+    }
+
+    lane.id
+  }
+
+  def updateLane(lane: PersistedLane, username: String ): Unit = {
+
+    val oldLaneCode = sql"""Select lane_code FROM LANE WHERE id = ${lane.id}""".as[Int].first
+
+    if ( MAIN_LANES.contains(oldLaneCode) && oldLaneCode != lane.laneCode )
+      throw new IllegalArgumentException("Cannot change the code of main lane!")
+
+    sqlu"""UPDATE LANE
+          SET LANE_CODE = ${lane.laneCode}, MODIFIED_BY = $username, MODIFIED_DATE = sysdate, MUNICIPALITY_CODE = ${lane.municipalityCode}
+          WHERE id = ${lane.id}
+    """.execute
+  }
+
+  def updateLanePosition(lane: PersistedLane, username: String ): Unit = {
+    sqlu"""UPDATE LANE_POSITION
+           SET SIDE_CODE = ${lane.sideCode}, START_MEASURE = ${lane.startMeasure}, END_MEASURE = ${lane.endMeasure}, LINK_ID = ${lane.linkId}
+          WHERE ID = (SELECT LANE_POSITION_ID FROM LANE_LINK WHERE LANE_ID = ${lane.id})
+     """.execute
+  }
+
+  def updateLaneAttributes(laneId: Long, props: LaneProperty, username: String ): Unit = {
+
+    val finalValue = if (props.values.isEmpty) "Null"
+                     else props.values.head.value.toString
+
+    sqlu"""MERGE INTO LANE_ATTRIBUTE
+          USING dual
+          ON ( name = ${props.publicId} AND LANE_ID =  $laneId )
+          WHEN MATCHED THEN
+            UPDATE SET VALUE = $finalValue, MODIFIED_BY = $username, MODIFIED_DATE = sysdate
+            WHERE LANE_ID =  $laneId
+            AND NAME = ${props.publicId}
+          WHEN NOT MATCHED THEN
+            INSERT (id, lane_id, name, value, created_date, created_by)
+            VALUES( ${Sequences.nextPrimaryKeySeqValue}, $laneId, ${props.publicId}, $finalValue, sysdate, $username)
+    """.execute
+  }
+
+
+  def updateLaneModifiedFields (laneId: Long, username: String) = {
+    sqlu"""
+      UPDATE LANE
+      SET modified_by = $username,
+          modified_date = SYSDATE
+      WHERE id = $laneId
+    """.execute
+  }
+
+  def updateMValues(id: Long, linkMeasures: (Double, Double), username: String): Unit = {
+    val (startMeasure, endMeasure) = linkMeasures
+
+    sqlu"""UPDATE LANE_POSITION
+           SET  START_MEASURE = $startMeasure, END_MEASURE = $endMeasure,  modified_date = SYSDATE
+          WHERE ID = (SELECT LANE_POSITION_ID FROM LANE_LINK WHERE LANE_ID = $id )
+     """.execute
+
+    updateLaneModifiedFields(id, username)
+  }
+
+  def updateMValuesChangeInfo(laneId: Long, linkMeasures: (Double, Double), vvhTimestamp: Long, username: String): Unit = {
+
+    val (startMeasure, endMeasure) = linkMeasures
+
+    sqlu"""
+      UPDATE LANE_POSITION
+      SET start_measure = $startMeasure, end_measure = $endMeasure, modified_date = SYSDATE, adjusted_timestamp = $vvhTimestamp
+      where id = (SELECT LANE_POSITION_ID FROM LANE_LINK WHERE LANE_ID = $laneId )
+    """.execute
+
+    updateLaneModifiedFields(laneId, username)
+  }
```

Comment:
We don't need to have these two similar methods only because of a different field.
We can do something like this:

```scala
  def updateMValues(id: Long, linkMeasures: (Double, Double), username: String, vvhTimestamp: Option[Long] = None): Unit = {
    val (startMeasure, endMeasure) = linkMeasures
    val vvhTimestampValue = vvhTimestamp.getOrElse(vvhClient.roadLinkData.createVVHTimeStamp())

    sqlu"""
         UPDATE LANE_POSITION
         SET start_measure = $startMeasure, end_measure = $endMeasure, modified_date = SYSDATE, adjusted_timestamp = $vvhTimestampValue
         WHERE ID = (SELECT LANE_POSITION_ID FROM LANE_LINK WHERE LANE_ID = $id )
     """.execute

    updateLaneModifiedFields(id, username)
  }